azurerm/internal/services/network/virtual_hub_connection_resource.go

```diff
@@ -170,36 +148,28 @@ func resourceArmVirtualHubConnectionRead(d *schema.ResourceData, meta interface{
 		return err
 	}
 
-	virtualHub, err := client.Get(ctx, id.ResourceGroup, id.VirtualHubName)
+	resp, err := client.Get(ctx, id.ResourceGroup, id.VirtualHubName, id.Name)
 	if err != nil {
-		if utils.ResponseWasNotFound(virtualHub.Response) {
-			log.Printf("[DEBUG] Virtual Hub %q was not found in Resource Group %q - so Connection %q can't exist - removing from state", id.Name, id.VirtualHubName, id.ResourceGroup)
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Virtual Hub Connection %q does not exist - removing from state", d.Id())
 			d.SetId("")
 			return nil
 		}
-
-		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): %+v", id.VirtualHubName, id.ResourceGroup, err)
+		return fmt.Errorf("reading Virtual Hub Connection %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
 	}
 
-	connection, err := findVirtualHubConnection(id.Name, virtualHub)
+	virtualHubResp, err := virtualHubClient.Get(ctx, id.ResourceGroup, id.VirtualHubName)
```

Comment:
we shouldn't need to look this up if we use an ID Formatter instead - can we switch this over?